includes/platform-checkout-user/class-platform-checkout-save-user.php

```diff
@@ -57,11 +58,17 @@ public function register_checkout_page_scripts() {
 	 * @return array
 	 */
 	public function maybe_add_userdata_to_metadata( $metadata, $order ) {
+		$session_data = WC()->session->get( 'platform-checkout-user-data' );
 		// If we should be creating a platform checkout customer, add the necessary customer data to the metadata.
-		$should_create_platform_customer = isset( $_POST['save_user_in_platform_checkout'] ) && filter_var( wp_unslash( $_POST['save_user_in_platform_checkout'] ), FILTER_VALIDATE_BOOLEAN ); // phpcs:ignore WordPress.Security.NonceVerification
-		if ( $should_create_platform_customer && ! empty( $_POST['platform_checkout_user_phone_field']['full'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
-				$platform_checkout_phone = wc_clean( wp_unslash( $_POST['platform_checkout_user_phone_field']['full'] ) ); // phpcs:ignore WordPress.Security.NonceVerification
+		$should_create_platform_customer = ( isset( $_POST['save_user_in_platform_checkout'] ) && filter_var( wp_unslash( $_POST['save_user_in_platform_checkout'] ), FILTER_VALIDATE_BOOLEAN ) ) || ( isset( $session_data['save_user_in_platform_checkout'] ) && filter_var( $session_data['save_user_in_platform_checkout'], FILTER_VALIDATE_BOOLEAN ) ); // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
Just a thought, but perhaps you could define this as a function somewhere--so we don't have to write this whole thing out each time--like perhaps in `Platform_Checkout_Utilities` maybe? Just seeing it used around quite a lot and it's a bit of a mouthful. 😬